tilequeue/command.py

```diff
@@ -1218,14 +1219,37 @@ def tilequeue_prune_tiles_of_interest(cfg, peripherals):
             ))
             for (x, y, z, tile_size, count) in cur:
                 coord = create_coord(x, y, z)
-                coord_int = coord_marshall_int(coord)
 
-                if not tile_size or tile_size == '256':
-                    # "uplift" a tile that is not explicitly a '512' size tile
-                    coord_int = coord_int_zoom_up(coord_int)
+                try:
+                    tile_size_as_zoom = metatile_zoom_from_str(tile_size)
+                    # tile size as zoom > cfg.metatile_zoom would mean that
+                    # someone requested a tile larger than the system is
+                    # currently configured to support (might have been a
+                    # previous configuration).
+                    assert tile_size_as_zoom <= cfg.metatile_zoom
+                    tile_zoom_offset = tile_size_as_zoom - cfg.metatile_zoom
+
+                except AssertionError:
```

Comment:
Would this be better as an `if` block rather than a `try/assert/except`?

Reply:
Yes, if that were the only assertion. Unfortunately, there's also one in `metatile_zoom_from_size`, called by `metatile_zoom_from_str`, which asserts that the metatile size is a power of two. I don't think it's possible to get a (200 OK) entry in the database like that, but I think we wouldn't want this process to stop in that case.

I could rewrite both functions to return `None` on error, and check for that. Do you think that would be better than the `try/except`?

Reply:
Ah, didn't see that. 👍

Reply:
Ah, didn't see the other assertion. 👍

Reply:
Catching an assertion error seems odd to me as well, and perhaps it would be better to return `None` instead and check for that at call sites.

By the way, it doesn't look like we're filtering on successful responses here, or I just missed it. Should we be?

Reply:
By successful responses, I just meant that we'd only consider 200 and 304 responses, because we could get garbage with 404s. But, the query looks like it'll guarantee that we always get valid coordinates, so on second thought maybe it's better to just leave the way it is. After all, we are interested in the requested coordinates here, regardless of what the response looked like, eg 5xx responses should be considered.

Never mind! :)